### PR TITLE
SwiftExpressionParser: simplify ASTWalker logic (NFC)

### PR DESCRIPTION
**Pull request by compnerd**
This adjusts the patterns used in the ASTWalker implementations to
reduce indentation and simplify the flow.  This is in preparation to
restructure the handling for default value thunks.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @dcci @xiaobai

**Comment by compnerd** [comment]
@swift-ci please test

File: source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp [hunk 1]
```diff
@@ -354,16 +350,15 @@ void SwiftASTManipulator::FindSpecialNames(
     SpecialNameFinder(NameVector &names, llvm::StringRef &prefix)
         : m_names(names), m_prefix(prefix) {}
 
-    virtual std::pair<bool, swift::Expr *> walkToExprPre(swift::Expr *expr) {
-      if (swift::UnresolvedDeclRefExpr *decl_ref_expr =
-              llvm::dyn_cast<swift::UnresolvedDeclRefExpr>(expr)) {
-        swift::Identifier name = decl_ref_expr->getName().getBaseIdentifier();
+    std::pair<bool, swift::Expr *> walkToExprPre(swift::Expr *E) override {
+      if (auto *DRE = llvm::dyn_cast<swift::UnresolvedDeclRefExpr>(E)) {
```
Comment (bulbazord) [inline source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp]: `DRE` -> `UDRE`? 😛

File: source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp [hunk 2]
```diff
@@ -493,49 +488,47 @@ bool SwiftASTManipulator::RewriteResult() {
...
-      swift::ReturnStmt *possible_return =
-          swift::dyn_cast<swift::ReturnStmt>(stmt);
-      if (possible_return && possible_return->hasResult()) {
-        swift::Expr *return_expr = possible_return->getResult();
-        if (return_expr) {
-          const bool add_return = true;
-          swift::Stmt *return_stmt;
+    swift::Stmt *walkToStmtPost(swift::Stmt *S) override {
+      auto *RS = swift::dyn_cast<swift::ReturnStmt>(S);
+      if (!RS)
```
Comment (bulbazord) [inline source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp]: You could turn this into `if (!RS || !RS->hasResult())` and get rid of the if below.

File: source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp [hunk 3]
```diff
@@ -493,49 +488,47 @@ bool SwiftASTManipulator::RewriteResult() {
...
+        return S;
 
-          return_stmt = m_manipulator.ConvertExpressionToTmpReturnVarAccess(
-              return_expr, possible_return->getStartLoc(), add_return,
-              m_decl_context);
-          if (return_stmt)
-            stmt = return_stmt;
+      if (RS->hasResult()) {
+        if (swift::Expr *RE = RS->getResult()) {
+          const bool add_return = true;
```
Comment (bulbazord) [inline source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp]: Something I think looks nicer, feel free to ignore:

You could fold `true` into the call below and add a small comment like so:
`m_manipulator.ConvertExpressionToTmpReturnVarAccess(RE, RS->getStartLoc(), /*add_return*/ true, m_decl_context)`

**Review by bulbazord — COMMENTED** [review]
A few things here or there but fine overall. I'd have @dcci look at it first though.

**Comment by dcci** [comment]
@swift-ci test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@dcci LGTM'ed this offline

**Pull request merged**